### ISSUE TITLE
Make -XX:doNotProcessJitEnvVars the default?

### ISSUE THREAD

**Issue by JamesKingdon** [eclipse-openj9/openj9#18831]
From time to time we get crashes in the libc getenv function caused by concurrent updates from third party libraries. The calls to getenv are frequently from the JIT since we have an abundant number of env vars that we query. These calls can be suppressed by using
-XX:doNotProcessJitEnvVars
Since it is unlikely that any of these env vars are used in production systems, could we make the above the default behaviour? This would reduce a pain point for our customers who are usually unaware that some third party code they are using is breaking the rules.

**Comment by pshipton** [comment]
@hzongaro fyi

**Comment by hzongaro** [comment]
@JamesKingdon, I agree that this is a pain point.  I ran into one of these late last year in issue #18640.

One difficulty might be in running tests where it can sometimes be difficult to pass `-Xjit` or `-Xaot` options or commands like `javac`, where those options are not supported.  In those cases, the only recourse for requesting trace information might be the `TR_Options` environment variable - but if an `-XX:processJitEnvVars` option can't be specified, at least not easily, that shuts off access to `TR_Options` in those cases.

Any thoughts about how we might get past that problem?

**Comment by pshipton** [comment]
Can you read a single environment variable once at startup to control the behavior of `-XX:doNotProcessJitEnvVars` ?

**Comment by jdmpapin** [comment]
> One difficulty might be in running tests where it can sometimes be difficult to pass `-Xjit` or `-Xaot` options or commands like `javac`, where those options are not supported.

Commands like `javac` can pass through arguments to the JVM using `-J<arg>`, e.g.
```
$ javac -J-Xjit:... Foo.java
```
